compiler/src/dotty/tools/io/Path.scala

```diff
@@ -127,7 +127,15 @@ class Path private[io] (val jpath: JPath) {
    * @return The path of the parent directory, or root if path is already root
    */
   def parent: Directory = jpath.normalize.getParent match {
```

Comment:
Perhaps it's best to avoid `normalize` altogether.

Consider this case:
- working directory is `/src/dotty`
- `foo` is a symlink to `/tmp/foo`

Before this PR:

```scala
scala> Path("foo/../bar").parent
val res0: dotty.tools.io.Directory = foo/../bar

scala> res0.jpath.toRealPath()
val res1: java.nio.file.Path = /tmp/bar
```

With this PR:

```scala
scala> Path("foo/../bar").parent
val res0: dotty.tools.io.Directory = .

scala> res0.jpath.toRealPath()
val res1: java.nio.file.Path = /src/dotty
```

Scala 2:

```scala
scala> Path("foo/../bar").parent
val res0: scala.reflect.io.Directory = foo/..

scala> res0.jfile.getCanonicalPath()
val res1: String = /tmp
```

readlink:

```
/src/dotty$ readlink -f foo/../bar/..
/tmp
```